### ISSUE TITLE
The regex returned by the `semver-regex` package produces different results from Node

### ISSUE THREAD

**Issue by vjpr** [oven-sh/bun#3809]
### What steps can reproduce the bug?

```ts
import semverRegex from 'semver-regex'
//const re = semverRegex()
const re = /(?<=^v?|\sv?)(?:(?:0|[1-9]\d{0,9}?)\.){2}(?:0|[1-9]\d{0,9})(?:-(?:--+)?(?:0|[1-9]\d*|\d*[a-z]+\d*)){0,100}(?=$| |\+|\.)(?:(?<=-\S+)(?:\.(?:--?|[\da-z-]*[a-z-]\d*|0|[1-9]\d*)){1,100}?)?(?!\.)(?:\+(?:[\da-z]\.?-?){1,100}?(?!\w))?(?!\+)/
const res = re.exec('v1.0.1448')
console.log({res})
```

### What is the expected behavior?

Node.js produces:

```
{
  res: [ '1.0.1448', index: 1, input: 'v1.0.1448', groups: undefined ]
}
```

### What do you see instead?

Bun:

```
{
  res: null
}
```

### Additional information

https://regex101.com/r/xTU6U9/1

**Comment by vjpr** [comment]
<img width="1676" alt="Screenshot 2023-07-25 at 8 24 00 pm" src="https://github.com/oven-sh/bun/assets/281413/9ff737fe-4933-4111-8ef3-ca731dd3b71c">

From Safari...looks like a JSC bug.

**Comment by dylan-conway** [comment]
The first lookbehind `(?<=^v?|\sv?)` seems to be part of the issue. Remove `^v?|`, `|\sv?`, or the entire lookbehind and the regex will behave normally.

**Comment by PaddeK** [comment]
Safari supports this only since 16.4 (Mar. 2023) [caniuse](https://caniuse.com/js-regexp-lookbehind)

**Comment by Jarred-Sumner** [comment]
Filed a bug in JSC

https://bugs.webkit.org/show_bug.cgi?id=259536

**Comment by Electroid** [comment]
As of Bun v1.0.7, this now produces a SyntaxError.

```js
error: Syntax Error!!
const re = /(?<=^v?|\sv?)(?:(?:0|[1-9]\d{0,9}?)\.){2}(?:0|[1-9]\d{0,9})(?:-(?:--+)?(?:0|[1-
           ^
```

After adding `// @bun` to the top of the file, the error straight from JSC is:

```js
SyntaxError: Unterminated regular expression literal '/(?<=^v?|\sv?)(?:(?:0|[1-9]\d{0,9}?)\.){2}(?:0|[1-9]\d{0,9})(?:-(?:--+)?(?:0|[1-9]\d*|\d*[a-z]+\d*)){0,100}(?=$| '
```

**Comment by dylan-conway** [comment]
Fixed in v1.0.3
